source/slang/ir.cpp

```diff
@@ -3737,8 +3794,46 @@ namespace Slang
         context->shared = &sharedContextStorage;
         context->builder = &sharedContextStorage.builderStorage;
         context->target = target;
-
-
+        Substitutions * curTailSubst = nullptr;
+        for (auto param : programLayout->genericEntryPointParams)
+        {
+            auto paramSubst = new GlobalGenericParamSubstitution();
+            if (!context->subst)
+                context->subst = paramSubst;
+            if (curTailSubst)
+                curTailSubst->outer = paramSubst;
+            curTailSubst = paramSubst;
+            paramSubst->paramDecl = param->decl;
```

Comment:
What is kind of unfortunate here (and I really can't see an obviously better option) is that you are building a distinct `Substitution` for *every parameter* rather than just one substitution that applies to all the global prameters.